app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -304,6 +304,7 @@ class SearchTEIViewModel(
                     }
                     SearchScreenState.MAP -> {
                         SearchIdlingResourceSingleton.increment()
+                        _refreshData.value = Unit
```

Comment:
Is there there the need to show the loading twice?. If so, would this trigger again the observable?